spectral_cube/dask_spectral_cube.py

```diff
@@ -0,0 +1,794 @@
+                              meta=meta,  unit=self.unit,
+                              header=header)
+
+        else:
+
+            return u.Quantity(out, unit=self.unit)
+
+    return wrapper
+
+
+class DaskSpectralCubeMixin:
+
+    def _compute(self, array):
+        # For now, always default to serial mode, but we could then expand this
+        # to allow different modes.
+        return array.compute(scheduler='synchronous')
+
+    @property
+    def _filled_dask_array(self):
+        if self._mask is None:
+            return self._data
+        else:
+            return da.asarray(self._mask._filled(data=self._data,
+                                                 wcs=self._wcs, fill=self.fill_value,
+                                                 wcs_tolerance=self._wcs_tolerance))
+
+    @property
+    def _nan_filled_dask_array(self):
+        if self._mask is None:
+            return self._data
+        else:
+            return da.asarray(self._mask._filled(data=self._data,
+                                                 wcs=self._wcs, fill=np.nan,
+                                                 wcs_tolerance=self._wcs_tolerance))
+
+    @warn_slow
+    @projection_if_needed
+    def sum(self, axis=None, **kwargs):
+        """
+        Return the sum of the cube, optionally over an axis.
+        """
+        return self._compute(da.nansum(self._nan_filled_dask_array, axis=axis))
+
+    @warn_slow
+    @projection_if_needed
+    def mean(self, axis=None, **kwargs):
+        """
+        Return the mean of the cube, optionally over an axis.
+        """
+        return self._compute(da.nanmean(self._nan_filled_dask_array, axis=axis))
+
+    @warn_slow
+    @projection_if_needed
+    def median(self, axis=None, **kwargs):
+        """
+        Return the median of the cube, optionally over an axis.
+        """
+        return self._compute(da.nanmedian(self._nan_filled_dask_array, axis=axis))
+
+    def percentile(self, q, axis=None, **kwargs):
+        """
+        Return percentiles of the data.
+
+        Parameters
+        ----------
+        q : float
+            The percentile to compute
+        axis : int, or None
+            Which axis to compute percentiles over
+        """
+        return self._compute(da.nanpercentile(self._nan_filled_dask_array, q, axis=axis))
+
+    @warn_slow
+    @projection_if_needed
+    def std(self, axis=None, ddof=0, **kwargs):
+        """
+        Return the mean of the cube, optionally over an axis.
+
+        Other Parameters
+        ----------------
+        ddof : int
+            Means Delta Degrees of Freedom.  The divisor used in calculations
+            is ``N - ddof``, where ``N`` represents the number of elements.  By
+            default ``ddof`` is zero.
+        """
+        return self._compute(da.nanstd(self._nan_filled_dask_array, axis=axis, ddof=ddof))
+
+    @warn_slow
+    @projection_if_needed
+    def mad_std(self, axis=None, **kwargs):
```

Comment:
`kwargs` aren't passed to `mad_std` below.

Reply:
Should `ignore_nan=True` by default? I guess we didn't have this in `SpectralCube` but seems like a reasonable default since the rest of the operations ignore nans, too.

I'll raise a separate issue.

Reply:
Fixed!